sdk/src/asset_handlers/pdf.rs

```diff
@@ -361,16 +408,28 @@ impl Pdf {
         Ok(())
     }
 
-    /// Sets the Associated File (/AF) key of the PDF to the provided embedded file spec reference.
-    fn set_af_relationship(&mut self, embedded_file_spec_ref: ObjectId) -> Result<(), Error> {
-        self.document
-            .catalog_mut()?
-            .set(ASSOCIATED_FILE_KEY, embedded_file_spec_ref);
+    /// Creates, or appends to, the Associated File (`AF`) array the embedded file spec reference of the
+    /// C2PA data.
+    fn push_associated_file(&mut self, embedded_file_spec_ref: ObjectId) -> Result<(), Error> {
+        let catalog = self.document.catalog_mut()?;
+        if catalog.get_mut(ASSOCIATED_FILE_KEY).is_err() {
```

Comment:
No big deal but there is a Hashmap function for this case of if entry not present create or else return the object.  It is called Vacanct and looks like this in practice:

if let std::collections::hash_map::Entry::Vacant(e) = self.claims_map.entry(label) {
                let index = self.claims.push_get_index(ingredient_claim);
                e.insert(index);
            }
Where e is the new entry or the existing entry you want to modify.

Reply:
nice, will take a look at that!